### PR TITLE
uaclient: add helper script for ec2 cleanup of stale VMs and images

### PR DESCRIPTION
**Pull request by blackboxsw**
This script cleans up vms, vpcs, security groups and images.

It will be called by a jenkins job running on torkoal per the following PR:
https://github.com/canonical/server-jenkins-jobs/pull/134

### CODE REVIEW

File: ubuntu-advantage-client/ec2_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        for internet_gateway in vpc.internet_gateways.filter(Filters=tag_filter):
+            print('terminating internet gateway %s' % internet_gateway.id)
+            internet_gateway.detach_from_vpc(VpcId=vpc.id)
+            internet_gateway.delete()
+
+        print('terminating vpc %s' % vpc.id)
+        vpc.delete()
+
+    print('# searching for ssh keys')
+    key_name = {'Name': 'key-name', 'Values': [CI_TAG]}
```
Comment (lucasmoura) [inline ubuntu-advantage-client/ec2_cleanup.py:47]: I think we should the `tag` parameter here instead

File: ubuntu-advantage-client/ec2_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    for key in client.describe_key_pairs(Filters=[key_name])['KeyPairs']:
+        print('deleting ssh key %s' % key['KeyName'])
+        client.delete_key_pair(KeyName=key['KeyName'])
+
+    print('# searching for amis')
+    for image in resource.images.filter(Owners=['self']).all():
```
Comment (lucasmoura) [inline ubuntu-advantage-client/ec2_cleanup.py:53]: Why is the images and snapshots not being deleted according to the tag ? It seems that we will delete all of them regardless of the tag associated to them

File: ubuntu-advantage-client/ec2_cleanup.py (hunk 1 above)
Comment (blackboxsw) [reply ubuntu-advantage-client/ec2_cleanup.py]: ahh yes I just saw that too. Will fix now

File: ubuntu-advantage-client/ec2_cleanup.py (hunk 2 above)
Comment (blackboxsw) [reply ubuntu-advantage-client/ec2_cleanup.py:66]: very good point will fix that now

Comment (blackboxsw) [reply ubuntu-advantage-client/ec2_cleanup.py:66]: passed tag_filter to this `images.filter` function now too
Can test with something like `./ubuntu-advantage-client/.tox/behave-aws/bin/python3 ubuntu-advantage-client/ec2_cleanup.py -t uaclient-ci-*`

**Pull request merged**